src/Providers/CloudWatchServiceProvider.php

```diff
@@ -43,20 +43,45 @@ public function boot()
     public function getLogger()
     {
         $cwClient = new CloudWatchLogsClient($this->getCredentials());
-        $loggingConfig = config('logging.channels.cloudwatch');
+        $loggingConfig = $this->app->make('config')->get('logging.channels.cloudwatch');
+
         $streamName = $loggingConfig['stream_name'];
         $retentionDays = $loggingConfig['retention'];
         $groupName = $loggingConfig['group_name'];
         $batchSize = isset($loggingConfig['batch_size']) ? $loggingConfig['batch_size'] : 10000;
+
         $logHandler = new CloudWatch($cwClient, $groupName, $streamName, $retentionDays, $batchSize);
         $logger = new Logger($loggingConfig['name']);
-        $formatter = new LineFormatter('%channel%: %level_name%: %message% %context% %extra%', null, false, true);
+
+        $formatter = $this->resolveFormatter($loggingConfig);
         $logHandler->setFormatter($formatter);
         $logger->pushHandler($logHandler);
 
         return $logger;
     }
 
+    /**
+     * Resolve a Formatter instance from configurations or use LineFormatter
+     * as default.
+     *
+     * @param array $configs
+     *
+     * @return \Monolog\Formatter\LineFormatter
+     */
+    private function resolveFormatter(array $configs)
+    {
+        $formatter = new LineFormatter(
+            '%channel%: %level_name%: %message% %context% %extra%',
+            null, false, true
+        );
+
+        if ($configs['formatter'] && class_exists($configs['formatter'])) {
+            $formatter = $this->app->make($configs['formatter']);
```

Comment:
it would look cleaner if you added a return here and returned the `LineFormatter` in the next line.
something like this
```
        if ($configs['formatter'] && class_exists($configs['formatter'])) {
              return $this->app->make($configs['formatter']);
        }

        return new LineFormatter(
            '%channel%: %level_name%: %message% %context% %extra%',
            null, false, true
        );
            
```

Reply:
Also `$configs['formatter']` will break when you do not define the `config`  for `cloudwatch`, if you do not add the key [here](https://github.com/pagevamp/laravel-cloudwatch-logs/blob/master/config/logging.php. ). Please add the key here as well and define a default. Since the default config cannot be defined as a string.
```
 new LineFormatter(
            '%channel%: %level_name%: %message% %context% %extra%',
            null, false, true
        );
```
May be add a callable check and pass the result as formatter.